app/backend/approaches/chatreadretrieveread.py

```diff
@@ -469,5 +376,57 @@ def get_source_file_with_sas(self, source_file: str) -> str:
             )
             return source_file + "?" + sas_token
         except Exception as error:
-            print(f"Unable to parse source file name: {str(error)}")
-            return ""
\ No newline at end of file
+            logging.error(f"Unable to parse source file name: {str(error)}")
+            return ""
+    
+    def get_prompt_template(self) -> PromptTemplate:      
```

Comment:
Why did this new PromptTemplate pattern need to be added? I assume the ungrounded approach file will simply be a new python file under the "approaches" folder.

Reply:
Yes, adding the prompt template class simplifies the approach once by pass RAG is allowed to be added.

Reply:
I am not sure how it simplifies the approach. The current RRR approach should be unchanged other than the enum change. A net new approach file should be added for Ungrounded to keep all the prompt engineering, calls out to other API services, etc. completely separate. So other than entry point there shouldn't be any need to modify RRR pattern.

Reply:
I believe its cleaner approach; I feel like we are splitting hairs here.     Instead of having two classes with the same class level variables, we now have a class that should be instantiated with shared properties.

Reply:
We are also balancing the fact that documentation, discussions, forked repos all have reference to this code that may need to be updated to keep compatibility. If the only change is to simply refactor this just to have it in a separate class but not provide any significant improvement then I am inclined to keep the pattern as it is. 

Also these definition of prompt templates would never be the same between multiple RAG patterns. For example, you would never need to use "query_prompt_template" in your ungrounded approach as you will never make a retrieval query against search. So trying to unify the "types" of prompt templates across all approaches will never align.

Please revert the change for "PromptTemplate()" and the rest can be approved. The refactoring of common methods to the "Approach" class is good and makes sense and the more declarative enum make the code more clear.

Reply:
PromptTemplate has been removed